src/TypeSpec.Extension/Emitter.Csharp/src/lib/clientModelBuilder.ts

```diff
@@ -113,6 +113,7 @@ export function createModelForService(
         Type: {
             Name: "String",
             Kind: InputTypeKind.String,
+            IsConfident: true,
```

Comment:
There are many places have such pattern like "we know it should be true so set true". So scattered, hard to maintain. Could we put all of them into same `getConfident` function?

Reply:
If you remember, in the first iteration of this PR, I want to calculate the `isConfident` in one place exactly in the way you are saying here, and you are the person that says we should calculate this property when we construct them.
I do not want to change it back

Reply:
Instead of changing back and forth, I am thinking that we should change our interfaces to classes so that it has constructors, and we could put those values in the constructor or the implementation of the property.
For instance, an issue the current design cannot resolve is that for the container types such as list and dictionary, the IsConfident should always redirect the value to its element type or value type, but now we cannot do that because during the calculation and the propagation, we always get to the container first, and then get to its element or value type. During this when we set the `IsConfident` of the element, we can no longer set the IsConfident of the container because it is very likely to be out of scope now (taking the `model Error { details: Error[] }` as an example to think this)